ironic/drivers/modules/drac/management.py

```diff
@@ -302,14 +315,182 @@ def set_boot_device(node, device, persistent=False):
 
 class DracRedfishManagement(redfish_management.RedfishManagement):
     """iDRAC Redfish interface for management-related actions.
-
-    Presently, this class entirely defers to its base class, a generic,
-    vendor-independent Redfish interface. Future resolution of Dell EMC-
-    specific incompatibilities and introduction of vendor value added
-    should be implemented by this class.
     """
-    pass
+    @METRICS.timer('DracRedfishManagement.clear_job_queue')
+    @base.clean_step(priority=0)
+    def clear_job_queue(self, task):
+        """Clear the job queue.
+
+        :param task: a TaskManager instance containing the node to act on.
+        :returns: None if it is completed.
+        :raises: DracOperationError on an error from python-dracclient.
+        """
+        system = redfish_utils.get_system(task.node)
+        for manager in system.managers:
+            try:
+                oem_manager = manager.get_oem_extension('Dell')
+            except sushy.exceptions.OEMExtensionNotFoundError as e:
+                error_msg = (_("Search for Sushy OEM extension package "
+                            "'sushy-oem-idrac' failed for node %(node)s. "
+                            "Ensure it is installed. Error: %(error)s") %
+                            {'node': task.node.uuid, 'error': e})
+                LOG.error(error_msg)
+                raise exception.RedfishError(error=error_msg)
+            try:
+                delete_job_response = oem_manager.clear_job_queue(
+                                                job_ids=['JID_CLEARALL'])
+            except sushy.exceptions.SushyError as e:
+                LOG.debug("Sushy OEM extension Python package "
+                          "sushy-oem-idrac failed to clear job queue"
+                          "for %(node)s, Error : %(error)s" %
+                          {'node':task.node.uuid, 'error':e})
+
+        if delete_job_response.status_code == _JOB_RESPONSE_CODE:
+            LOG.info("Job queue cleared for node %(node)s via OEM" %
+                    {'node': task.node.uuid})
+        else:
+            LOG.error("Failed to clear job queue, node : %(node)s " %
+                     {'node': task.node.uuid})
+        return delete_job_response
```

Comment:
Cleaning steps should either return states.CLEANWAIT or None.  See:
https://github.com/openstack/ironic/blob/master/ironic/drivers/base.py#L307
Since this cleaning step is synchronous, I think it should return None.

Reply:
Addressed.